feature/experimental/system/health/tests/system_generic_health_check/system_generic_health_check_test.go

```diff
@@ -0,0 +1,755 @@
+		timestamp := time.Now().Round(time.Second)
+
+		query := gnmi.OC().Component(cpu).State()
+		component := gnmi.Get(t, dut, query)
+
+		cpuParent := component.GetParent()
+		if cpuParent == "" {
+			t.Errorf("ERROR: can't find parent information for CPU card %v", component)
+		}
+
+		// If cpu card's parent is line card, check cpu ultilization.
+		if contains(cardList, cpuParent) {
+			// Remove parent from the list of check cards.
+			cardList = removeElement(cardList, cpuParent)
+			// Fetch CPU utilization data.
+			cpuUtilization := component.GetCpu().GetUtilization()
+			if cpuUtilization == nil {
+				t.Errorf("ERROR: %s %s %s Type %-20s - CPU utilization data not available",
+					timestamp, deviceName, cpu, cpuParent)
+				continue
+			}
+			averageUtilization := cpuUtilization.GetAvg()
+			if averageUtilization > cpuHighUtilization {
+				t.Errorf("ERROR: %s %s Type %-20s %-10s - Utilization: %3d%%, exceeding threshold: %3d%%",
+					timestamp, deviceName, cpu, cpuParent, averageUtilization, cpuHighUtilization)
+			} else {
+				t.Logf("INFO: %s %s Type %-20s %-10s - Utilization: %3d%%",
+					timestamp, deviceName, cpu, cpuParent, averageUtilization)
+			}
+		}
+	}
+
+	if len(cardList) > 0 {
+		t.Errorf("ERROR: Didn't find cpu card for checkCards %s", cardList)
+	}
+}
+
+func TestComponentsNoHighMemoryUtilization(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	deviceName := dut.Name()
+	description := "Component"
+
+	controllerCards := components.FindComponentsByType(t, dut, controllerCardType)
+	lineCards := components.FindComponentsByType(t, dut, lineCardType)
+	cardList := append(controllerCards, lineCards...)
+	for _, component := range cardList {
+		query := gnmi.OC().Component(component).State()
+		timestamp := time.Now().Round(time.Second)
+		componentState := gnmi.Get(t, dut, query)
+		componentType := componentState.GetType()
+		memoryState := componentState.GetMemory()
+		if memoryState == nil {
+			t.Errorf("ERROR: %s - Device: %s - %s: %-40s - Type: %-20s - Memory data not available",
+				timestamp, deviceName, description, component, componentType)
+			continue
+		}
+
+		memoryAvailable := memoryState.GetAvailable()
+		memoryUtilized := memoryState.GetUtilized()
+		memoryUtilization := uint8((memoryUtilized * 100) / (memoryAvailable + memoryUtilized))
+		if memoryUtilization > memoryHighUtilization {
+			t.Errorf("ERROR: %s - Device: %s - %s: %-40s - Type: %-20s - Utilization: %3d%% (Threshold: %3d%%)",
+				timestamp, deviceName, description, component, componentType, memoryUtilization, memoryHighUtilization)
+		} else {
+			t.Logf("INFO: %s - Device: %s - %s: %-40s - Type: %-20s - Utilization: %3d%%",
+				timestamp, deviceName, description, component, componentType, memoryUtilization)
+		}
+	}
+}
+
+func TestSystemProcessNoHighCPUSpike(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	deviceName := dut.Name()
+
+	query := gnmi.OC().System().ProcessAny().State()
+	description := "System CPU Process"
```

Comment:
Made threshold a const; description is used in different tests with different values; instead of making it a map string var up top, left it as a local variable in each test. Thanks!

Reply:
Thanks for your comments. Updated as a local const in Testcase:
```go
	const description = "System CPU Process"
```